app/ui/setup.py

```diff
@@ -172,3 +184,208 @@ def sndk_url(account: str, user: str, region: str) -> str:
+
+
+def get_api_gateway_url(sf_region: str, sd_deployment: str) -> str:
+
+    stage_url_map = {
+        "us-east-1": API_GATEWAY_STAGE_US_EAST_1,
+        "us-west-2": API_GATEWAY_STAGE_US_WEST_2,
+    }
+
+    prod_url_map = {
+        "us-east-1": API_GATEWAY_PROD_US_EAST_1,
+        "us-east-2": API_GATEWAY_PROD_US_EAST_2,
+        "us-west-2": API_GATEWAY_PROD_US_WEST_2,
+    }
+
+    if sd_deployment == "dev":
+        return API_GATEWAY_DEV_US_WEST_2
+    elif sd_deployment == "prod":
+        return prod_url_map[get_sundeck_region(sf_region)]
+    elif sd_deployment == "stage":
+        sundeck_region = get_sundeck_region(sf_region)
+        if sundeck_region == "us-east-1":
+            return stage_url_map[sundeck_region]
+        else:
+            return stage_url_map["us-west-2"]
+    else:
+        raise Exception("Invalid deployment")
+
+
+def get_sundeck_region(sf_region: str) -> str:
+    # Supported Sundeck Regions ["us-east-1", "us-east-2.aws", "us-west-2"]
+    # This is a static map of snowflake-region to nearby supported sundeck-region.
+    # This is used during creation of Sundeck account, to pick nearest sundeck-region.
+    sf2sd_region_map = {
+        "AWS_US_WEST_2": "us-west-2",
+        "AWS_US_EAST_1": "us-east-1",
+        "AWS_AP_SOUTHEAST_2": "us-west-2",
+        "AWS_EU_WEST_1": "us-east-1",
+        "AWS_AP_SOUTHEAST_1": "us-west-2",
+        "AWS_CA_CENTRAL_1": "us-west-2",
+        "AWS_EU_CENTRAL_1": "us-east-1",
+        "AWS_US_EAST_2": "us-east-2",
+        "AWS_AP_NORTHEAST_1": "us-west-2",
+        "AWS_AP_SOUTH_1": "us-west-2",
+        "AWS_EU_WEST_2": "us-east-1",
+        "AWS_AP_NORTHEAST_2": "us-west-2",
+        "AWS_EU_NORTH_1": "us-east-1",
+        "AWS_AP_NORTHEAST_3": "us-west-2",
+        "AWS_SA_EAST_1": "us-west-2",
+        "AWS_EU_WEST_3": "us-east-1",
+        "AWS_AP_SOUTHEAST_3": "us-west-2",
+        "AWS_US_GOV_WEST_1": "us-west-2",
+        "AWS_US_EAST_1_GOV": "us-east-1",
+    }
+
+    return sf2sd_region_map[sf_region]
+
+
+def get_region(sf_region: str) -> str:
+    # sf_region has the format AWS_<REGION>_<AZ>
+    region_map = {
+        "AWS_US_WEST_2": "us-west-2",
+        "AWS_US_EAST_1": "us-east-1",
+        "AWS_AP_SOUTHEAST_2": "ap-southeast-2",
+        "AWS_EU_WEST_1": "eu-west-1",
+        "AWS_AP_SOUTHEAST_1": "ap-southeast-1",
+        "AWS_CA_CENTRAL_1": "ca-central-1",
+        "AWS_EU_CENTRAL_1": "eu-central-1",
+        "AWS_US_EAST_2": "us-east-2",
+        "AWS_AP_NORTHEAST_1": "ap-northeast-1",
+        "AWS_AP_SOUTH_1": "ap-south-1",
+        "AWS_EU_WEST_2": "eu-west-2",
+        "AWS_AP_NORTHEAST_2": "ap-northeast-2",
+        "AWS_EU_NORTH_1": "eu-north-1",
+        "AWS_AP_NORTHEAST_3": "ap-northeast-3",
+        "AWS_SA_EAST_1": "sa-east-1",
+        "AWS_EU_WEST_3": "eu-west-3",
+        "AWS_AP_SOUTHEAST_3": "ap-southeast-3",
+        "AWS_US_GOV_WEST_1": "us-gov-west-2",
+        "AWS_US_EAST_1_GOV": "us-east-1",
+    }
```

Comment:
I find myself wishing these two maps were built from the same key-set. When Snowflake inevitably adds a new region, we would want to make sure both of these `dicts` are updated.

Could we create these dicts once and just expose the functions? I would love a comment to remind developers later to make sure both maps are updated (unless you can come up with a good way to make it hard for a developer to accidentally add a region to one of the dicts and not the other :) )